### PR TITLE
Adds training script 

### PR DESCRIPTION
**Pull request by ilyakava**
Hi and thank you for sharing the DPTNet code. I've added a training script that you will recognize from: https://github.com/ShiZiqiang/dual-path-RNNs-DPRNNs-based-speech-separation

The `solver.py`  is basically copy-pasted from that repo, as well as their `train.py` -> `train_and_eval.py` here.

I kept all your parameters the same in the default model. I also added a logger that prints and saves to file, and updated the readme with example runs. There are also a few more cmd line arguments added for resuming training from saved checkpoints.

I'm training this now, batch size 1 with wsj0-2mix on 1x3090 takes about 12 epochs per day. Is this in line with your training procedure?

I have found that larger batch size leads to worse performance, this means that it must be trained on 1 device. Is this also in line with your experience?

### CODE REVIEW

**Comment by ilyakava** [comment]
I was able to train this for 100 epochs once and got a val loss of `-20.156` on wsj0-2mix

**Comment by ilyakava** [comment]
I trained this a second time for 100 epochs and got training loss down to `-21.109` and eval loss down to `-20.229` on wsj0-2mix
